### ISSUE TITLE
Fix Background sometimes turning permanently bright yellow due to the image being used by "another process" randomly. (Currently bandaid-ed)

### ISSUE THREAD

**Issue by Untrustedlife** [Revolutionary-Games/Thrive#639]
So you have to play for a few generations, then die a couple times, and the background turns bright yellow and burns everyone's eyes out due to its brightness.

Here is the error log that is associated:
![unknown](https://user-images.githubusercontent.com/4032123/48593872-2ab62d00-e914-11e8-8d45-145ad4bff468.png)
![unknown 1](https://user-images.githubusercontent.com/4032123/48593873-2c7ff080-e914-11e8-99f3-47bfd56c7c1f.png)

**Comment by Untrustedlife** [comment]
It happened again:
![image](https://user-images.githubusercontent.com/4032123/48598213-5c84bf00-e927-11e8-9053-5713a5f24707.png)
[ThriveLogOGRE (1).txt](https://github.com/Revolutionary-Games/Thrive/files/2587984/ThriveLogOGRE.1.txt)

**Comment by Untrustedlife** [comment]
SO its defintly the background somehow getting loaded and not unloaded.

**Comment by Untrustedlife** [comment]
And it happened again:
![image](https://user-images.githubusercontent.com/4032123/48598586-28aa9900-e929-11e8-92c6-f6dfb2d8ddca.png)

**Comment by Untrustedlife** [comment]
Okay, probably same error. I dont know where it is hapenning

**Comment by Untrustedlife** [comment]
what it is is the baclkground being opened up by the game and not "closed" when you die

**Comment by Untrustedlife** [comment]
I think.. But it only happens sometimes.

**Comment by Untrustedlife** [comment]
Perhaps dying and entering editor, then dying again is what breaks it?

**Comment by Untrustedlife** [comment]
Now that i think about it, thats most definitely whats happening

**Comment by crodnu** [comment]
maybe once a file is opened it cannot be opened ever again? because it's never closed?

**Comment by Untrustedlife** [comment]
So, you go to the editor right, well then you die, well i bet the editor isnt closing the background so you switch biomes and it looks fine to you but then you die and it switches back to your old biome, but its still "open" and when it fails to grab it ., tahts when big yellow happens

**Comment by Untrustedlife** [comment]
A bandaid would be just to make the editor use its own background But we would rather it be fixed

**Comment by crodnu** [comment]
things like this is why we don't use new and delete anymore in c++ smh

**Comment by Untrustedlife** [comment]
I added an if statement in the editor creation function (which was calling create new background every time it ran and i think it fixed it

**Comment by Untrustedlife** [comment]
https://github.com/Revolutionary-Games/Thrive/commit/0cf6f1b026a49a044cf11467d0d42d65d8680214

This is the change that seems to have fixed it

**Comment by Untrustedlife** [comment]
Im not going to close it just yet, i would rather others try it out and confirm its fixed aswell.

**Comment by Untrustedlife** [comment]
So basically instead of creating a new one each time you enter the editor now it creates it once and setbackground and all that still work, so the editor still changes to the current biomes background but doesn't try loading the texture multiple times which locked it out.
Because we didnt need to create it each time like we were doing since setBackground already altered both.

**Comment by Untrustedlife** [comment]
:)

**Comment by Untrustedlife** [comment]
If there is no objection ill close this when i wake up in the morning, that was a conundrum of an issue.

**Comment by Untrustedlife** [comment]
The new ogre verison diidn t fix it, so we have to use this fix for now,

**Comment by Untrustedlife** [comment]
i edited the title and added it to milestone 0.4.1, the bandaid works, for now

**Comment by hhyyrylainen** [comment]
Okay, so there still is a problem? It is just very rare.

**Comment by Untrustedlife** [comment]
I haven’t seen it after the change I made last night. But that doesn’t mean it’s fixed.

**Comment by Untrustedlife** [comment]
And I tested quite a bit last night.

**Comment by Untrustedlife** [comment]
So I think we should leave this up. If people play a bunch and don’t encounter the issue after i putbout a new indev version today before the actual 0.4.0 release day (whenever that is) then it’s probably fixed.

**Comment by hhyyrylainen** [comment]
If this gets closed before the underlying issue is fixed I'll just open a new issue for tracking that.

**Comment by Untrustedlife** [comment]
Ayy. i think this is a specific case. But whatever you feel is correct

**Comment by Untrustedlife** [comment]
So it happened again, after running several tests (only once), so its still happening its just alot rarer, im going to try one more thing, im going to seperate the editor background mesh and the normal background mesh

**Comment by Untrustedlife** [comment]
Are we sure we cant force ogre to close the texture

**Comment by hhyyrylainen** [comment]
We never open the texture ourselves, it all happens internally by us just setting the material name. That's the reason why I think this is an Ogre bug.

**Comment by hhyyrylainen** [comment]
I hope this is now fixed with the increased file descriptor limit (on Windows) and the change to the audio library by @crodnu to reduce the open file count.
